clippy_lints/src/utils/mod.rs

```diff
@@ -634,10 +634,45 @@ impl<'a> DiagnosticWrapper<'a> {
     }
 }
 
+/// Emit a basic lint message with a `msg` and a `span`.
+///
+/// This is the most primitive of our lint emission methods and can
```

Comment:
Is 'lint emission methods' a good term for these functions? I can't really think of a better one.